File: jdroid-android-sample/src/main/java/com/jdroid/android/sample/ui/service/SampleWorker4.java
package com.jdroid.android.sample.ui.service;

import android.content.Context;
import android.support.annotation.NonNull;

import com.jdroid.android.jetpack.work.AbstractWorker;
import com.jdroid.java.concurrent.ExecutorUtils;

import java.util.concurrent.TimeUnit;

import androidx.work.Worker;
import androidx.work.WorkerParameters;

public class SampleWorker4 extends AbstractWorker {

	public SampleWorker4(@NonNull Context context, @NonNull WorkerParameters workerParams) {
		super(context, workerParams);
	}

	@NonNull
	@Override
	protected Worker.Result onWork() {
		ExecutorUtils.sleep(20, TimeUnit.SECONDS);
		return Result.SUCCESS;
	}
	
	// TODO
//	@Nullable
//	@Override
//	protected Long getMinimumTimeBetweenExecutions() {
//		return TimeUnit.MINUTES.toMillis(1);
//	}
}

File: jdroid-android-sample/src/main/java/com/jdroid/android/sample/ui/service/SampleWorker4.kt
package com.jdroid.android.sample.ui.service

import android.content.Context
import androidx.work.WorkerParameters
import com.jdroid.android.jetpack.work.AbstractWorker
import com.jdroid.java.concurrent.ExecutorUtils.sleep
import java.util.concurrent.TimeUnit

class SampleWorker4(context: Context, workerParams: WorkerParameters) :
    AbstractWorker(context, workerParams) {
    override fun onWork(): Result {
        sleep(20, TimeUnit.SECONDS)
        return Result.success()
    }

    // TODO
    //	@Nullable
    //	@Override
    //	protected Long getMinimumTimeBetweenExecutions() {
    //		return TimeUnit.MINUTES.toMillis(1);
    //	}
}